subsys/bootloader/CMakeLists.txt

```diff
@@ -17,11 +17,12 @@ if (NOT IMAGE_NAME)
   zephyr_include_directories(include/dummy_values/)
 endif()
 
-if (DEFINED CONFIG_SOC_NRF9160)
+if (DEFINED CONFIG_SOC_NRF9160 OR DEFINED CONFIG_SOC_NRF5340_CPUAPP)
   message(WARNING "
       --------------------------------------------------------
       --- WARNING: When using the immutable bootloader on  ---
-      --- nRF9160, the UICR must be erased when flashing.  ---
-      --- E.g. by calling 'west flash --erase'             ---
+      --- nRF9160 or nRF5340 application core, the UICR    ---
```

Comment:
Could we drop the names of the specific boards here, since the `if DEFINED` already checks it?

Something like `The immutable bootloader stores data in UICR, hence the UICR must be deleted before flashing the immutable bootloader` ...

Reply:
Changed to "this SoC"